src/py-opentimelineio/opentimelineio-bindings/otio_utils.cpp

```diff
@@ -94,30 +95,126 @@ void _build_any_to_py_dispatch_table() {
     }
 }
 
-static py::object _value_to_any = py::none();
+any py_to_any(py::handle const& o) {
+    if (o.ptr() == nullptr || o.is_none()) {
+        return any(nullptr);
+    }
+
+    if (py::isinstance<py::bool_>(o)) {
+        return any(py_to_cpp(py::cast<py::bool_>(o)));
+    }
+
+    if (py::isinstance<py::int_>(o)) {
+        try {
+            return any(py_to_cpp<std::int32_t>(py::cast<py::int_>(o)));
+        } catch (...) {}
+
+        try {
+            return any(py_to_cpp<std::int64_t>(py::cast<py::int_>(o)));
+        } catch (...) {}
+
+        try {
+            return any(py_to_cpp<std::uint32_t>(py::cast<py::int_>(o)));
+        } catch (...) {}
+
+        try {
+            return any(py_to_cpp<std::uint64_t>(py::cast<py::int_>(o)));
+        } catch (...) {}
+
+        throw py::type_error("Failed to convert Python int to C++ int");
+    }
+
+    if (py::isinstance<py::float_>(o)) {
+        return any(py_to_cpp(py::cast<py::float_>(o)));
+    }
+
+    if (py::isinstance<py::str>(o)) {
+        return any(py_to_cpp(py::cast<py::str>(o)));
+    }
+
+    // Convert AnyDictionaryProxy and dict before vector and sequence because
+    // a dict is a sequence.
+    if (py::isinstance<AnyDictionaryProxy>(o)) {
+        return any(o.cast<AnyDictionaryProxy>().fetch_any_dictionary());
+    }
+
+    if (py::isinstance<py::dict>(o)) {
+        return any(py_to_cpp(py::cast<py::dict>(o)));
+    }
+
+    if (py::isinstance<AnyVectorProxy>(o)) {
+        return any(o.cast<AnyVectorProxy>().fetch_any_vector());
+    }
+
+    if (py::isinstance<py::sequence>(o)) {
+        return any(py_to_cpp(py::cast<py::iterable>(o)));
+    }
 
-static void py_to_any(py::object const& o, any* result) {
-    if (_value_to_any.is_none()) {
-        py::object core = py::module::import("opentimelineio.core");
-        _value_to_any = core.attr("_value_to_any");
+    if (py::isinstance<RationalTime>(o)) {
+        return any(py_to_cpp<RationalTime>(o));
     }
 
-    result->swap(_value_to_any(o).cast<PyAny*>()->a);
+    if (py::isinstance<TimeRange>(o)) {
+        return any(py_to_cpp<TimeRange>(o));
+    }
+
+    if (py::isinstance<TimeTransform>(o)) {
+        return any(py_to_cpp<TimeTransform>(o));
+    }
+
+    if (py::isinstance<SerializableObject>(o)) {
+        SerializableObject::Retainer<> r(py::cast<SerializableObject*>(o));
+        return create_safely_typed_any(r.take_value());
```

Comment:
I have to admit that I'm not too sure how the downcast works here, but it works. This is taken from https://github.com/AcademySoftwareFoundation/OpenTimelineIO/blob/main/src/py-opentimelineio/opentimelineio-bindings/otio_bindings.cpp#L31-L40.

We'll need more tests to make sure that everything works as expected.